src/phantom_token.c

```diff
@@ -512,11 +577,15 @@ static ngx_int_t introspection_response_handler(ngx_http_request_t *request, voi
 
     module_context->status = request->headers_out.status;
 
-    // fail early for not 200 response
+    // Fail early for not 200 response
     if (request->headers_out.status != NGX_HTTP_OK)
     {
+        // Log any error except 204 expiry responses, which are expected
+        if (request->headers_out.status != NGX_HTTP_NO_CONTENT) {
```

Comment:
There were a few places where unexpected errors wrote nothing to logs and the caller got a 502.
Now, we write a log entry so that it is clear the phantom token module failed.
Note that I do not log token 401s since they were not logged before.